File: app/src/main/java/com/raju/javabaseproject/data/source/remote/utils/interceptors/OfflineInterceptor.java
package com.raju.javabaseproject.data.source.remote.utils.interceptors;

import android.content.Context;

import com.raju.javabaseproject.data.source.remote.utils.NetworkUtil;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

import okhttp3.CacheControl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class OfflineInterceptor implements Interceptor {

    private Context context;

    public OfflineInterceptor(Context context) {
        this.context = context;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        if (!NetworkUtil.isConnected(context)) {
            CacheControl cacheControl = new CacheControl.Builder()
                    .maxStale(7, TimeUnit.DAYS)
                    .build();
            request = request.newBuilder()
                    .cacheControl(cacheControl)
                    .build();
        }

        return chain.proceed(request);
    }
}

File: app/src/main/java/com/raju/javabaseproject/data/source/remote/http/OfflineInterceptor.kt
package com.raju.javabaseproject.data.source.remote.http

import android.content.Context

import com.raju.javabaseproject.data.utils.NetworkUtil

import java.io.IOException
import java.util.concurrent.TimeUnit

import okhttp3.CacheControl
import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response


class OfflineInterceptor(private val context: Context) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        var request = chain.request()
        if (!NetworkUtil.isConnected(context)) {
            val cacheControl = CacheControl.Builder()
                    .maxStale(7, TimeUnit.DAYS)
                    .build()
            request = request.newBuilder()
                    .cacheControl(cacheControl)
                    .build()
        }

        return chain.proceed(request)
    }
}

